cdap-api/src/main/java/co/cask/cdap/api/mapreduce/AbstractMapReduce.java

```diff
@@ -125,27 +112,18 @@ public final void initialize(MapReduceContext context) throws Exception {
   /**
    * Classes derived from {@link AbstractMapReduce} can override this method to initialize the {@link MapReduce}.
    * {@link MapReduceContext} will be available in this method using {@link AbstractMapReduce#getContext}.
-   * Default implementation of this method calls the deprecated {@link AbstractMapReduce#beforeSubmit} method.
    * @throws Exception if there is any error in initializing the MapReduce
    */
   @TransactionPolicy(TransactionControl.IMPLICIT)
   protected void initialize() throws Exception {
-    beforeSubmit(context);
```

Comment:
I believe it's a warning to have an empty method. Replace these with the 'Do nothing by default' comment